### DISCUSSION TITLE
WebUI email/password sign in broken in most recent docker image

### DISCUSSION THREAD

**Discussion by dannystaple** [open-webui/open-webui#4707] [General]
# Bug Report

## Installation Method

[Describe the method you used to install the project, e.g., git clone, Docker, pip, etc.]
Installation using docker, with the following script:

```bash
docker run -d \
  -p 3000:8080 \
  --add-host=host.docker.internal:host-gateway \
  -v /home/danny/ai/webui-settings:/app/backend/data \
  -e WEBUI_AUTH=True \
  --name open-webui \
  --restart always ghcr.io/open-webui/open-webui:main
```

- works: git-8d257ed
- doesn't: git-13b0e7d

## Environment

- **Open WebUI Version:** - git-8d257ed (docker images are tagged main, latest and git, but not webui versions) - 0.3.13.
- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04] - Docker
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0] - All

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

[Describe what you expected to happen.]

Auth continues to work in the Latest/main docker image.

## Actual Behavior:

[Describe what actually happened.]

In docker image created 4 days ago - tagged git-13b0e7d, created in job https://github.com/open-webui/open-webui/actions/runs/10393992869, auth no longer works, all users are unable to autheticate.

## Description

**Bug Summary:**
[Provide a brief but clear summary of the bug]

User authentication breaks, every user is told incorrect credentials, after starting WebUI with an existing set of data, but this newer image. Resetting to the previous docker image tag works.

## Reproduction Details

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

- Set up an auth WebUI using git-8d257ed. Start up with a bind mount for data.
- Log in, create users with the admin panel.
- Update the image to the current main or git-13b0e7d.
- Try to log in again.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]
Logging doesn't show anything.

**Screenshots/Screen Recordings (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by dannystaple** [comment]
Good morning, why has this been moved to a discussion please?

**Reply by justinh-rahb** [reply]
Multiple devs have been unable to reproduce this, and your report lacks enough information for us to infer an error otherwise.

**Comment by dannystaple** [comment]
Understood. I am talking here about the internal auth using bcrypted passwords in the sqllite database, with the ` WEBUI_AUTH=True` setting enabled. This is authenticating to the web client using an email address/password.
I'm presuming it's the path through https://github.com/open-webui/open-webui/blob/ec99ac71214c4866381f6005627711e4d1f2e10f/backend/apps/webui/routers/auths.py#L124.

I was unable to sign into the web Ui on that version. I've currently pinned mine to  git-8d257ed, which does allow me to sign in.

**Reply by dannystaple** [reply]
One thought would be to patch a version in a container with the exception in https://github.com/open-webui/open-webui/blame/ec99ac71214c4866381f6005627711e4d1f2e10f/backend/apps/webui/models/auths.py#L129 temporarily set up to log.exception, which may reveal a stack trace on why it's failing (and doesn't in the other version). the container image git-13b0e7d should have the code tagged 0.3.13.

**Reply by dannystaple** [reply]
This stack trace (shoulda posted it):
```python
(trapped) error reading bcrypt version
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/passlib/handlers/bcrypt.py", line 620, in _load_backend_mixin
    version = _bcrypt.__about__.__version__
              ^^^^^^^^^^^^^^^^^
AttributeError: module 'bcrypt' has no attribute '__about__'
```

This looks related to https://github.com/pyca/bcrypt/issues/684.
